x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlInitializationService.java

```diff
@@ -177,19 +177,27 @@ private void makeMlInternalIndicesHidden() {
             IndicesAliasesRequest indicesAliasesRequest = new IndicesAliasesRequest();
             for (ObjectObjectCursor<String, List<AliasMetadata>> entry : getAliasesResponse.getAliases()) {
                 String index = entry.key;
-                String[] nonHiddenAliases = entry.value.stream()
-                    .filter(metadata -> metadata.isHidden() == null || metadata.isHidden() == false)
-                    .map(AliasMetadata::alias)
-                    .toArray(String[]::new);
-                if (nonHiddenAliases.length == 0) {
-                    continue;
-                }
-                indicesAliasesRequest.addAliasAction(
-                    IndicesAliasesRequest.AliasActions.add()
+                for (AliasMetadata existingAliasMetadata : entry.value) {
+                    if (existingAliasMetadata.isHidden() == null || existingAliasMetadata.isHidden() == false) {
```

Comment:
FYI: This condition must be reversed. I did that in my PR (https://github.com/elastic/elasticsearch/pull/80346) which replaces this one.

Reply:
Oh dear.  I think that invalidates the testing we did on https://github.com/elastic/elasticsearch/pull/79123.  Please could reopen that PR, apply the fix, and run a few more BWC test jobs that will do real upgrades of 6.8 to 7.16.  Assuming they all work that PR can then be closed again.

This was my mistake.  Sorry for the inconvenience.